src/compiler/checker.ts

```diff
@@ -13861,7 +13861,8 @@ namespace ts {
                     }
                     else if (noImplicitAny && !compilerOptions.suppressImplicitAnyIndexErrors && !suppressNoImplicitAnyError) {
                         if (propName !== undefined && typeHasStaticProperty(propName, objectType)) {
-                            error(accessExpression, Diagnostics.Property_0_is_a_static_member_of_type_1, propName as string, typeToString(objectType));
+                            const typeName = typeToString(objectType);
+                            error(accessExpression, Diagnostics.Property_0_does_not_exist_on_type_1_Did_you_mean_to_access_the_static_member_2_instead, propName as string, typeName, typeName + '["' + propName + '"]');

```

Comment:
You'll need to be a little careful on this one in case of dumb inputs. For example,

```
class C {
  static "\""() { }
}

(new C())["\""]();
```

Reply:
Pretty sure we have other places in the checker or display builder that does similar logic.

Reply:
@DanielRosenwasser Thanks. I changed that.

Reply:
@DanielRosenwasser Thanks. I changed that and added a test to cover this case.

Reply:
@DanielRosenwasser Thanks. I changed that and added a test to cover this case.

Reply:
@DanielRosenwasser Do you have any additional cases which need to cover?